Docs 12259 d wv11xrefs duke (#52)
mulesoft/docs-dataweave: modules/ROOT/pages/dataweave-flat-file-schemas.adoc

include::partial$eol-warning.adoc[]

> xref:index.adoc[DataWeave] allows you to process several different xref:dataweave-formats.adoc[types of data]. For most of these types, you can import a schema that describes the input structure in order to have access to valuable metadata at design time. See xref:6@studio::input-output-structure-transformation-studio-task.adoc[To Define Input and Output Structure of a Transformation].

Author: ```suggestion
DataWeave enables you to process several different xref:dataweave-formats.adoc[types of data]. For most of these types, you can import a schema that describes the input structure in order to have access to valuable metadata at design time. See xref:6@studio::input-output-structure-transformation-studio-task.adoc[To Define Input and Output Structure of a Transformation].
```

=== Multiple Structures

> If you have multiple structures or segment definitions in an FFD, when you apply your schema to an metadata description on a xref:index.adoc[Transform Message component] you'll need to specify which one you want to use.

Author: ```suggestion
If you have multiple structures or segment definitions in an FFD, when you apply your schema to an metadata description on a Transform Message component you'll need to specify which one you want to use.
```

. Download the sample schema file link:{attachmentsdir}/Check.ffd[here].

. In Anypoint Studio click *File* > *New* > *Mule Project* to create a new project, give it whatever name you wish and click *Finish*.

> . Drag an HTTP Connector from the palette to your empty canvas, then a *Transform Message Component*.

Author: ```suggestion
. Drag an HTTP Connector from the palette to your empty canvas, then a Transform Message Component.
```

> . Drag an xref:3.8@mule-runtime::http-connector.adoc[HTTP Connector] from the palette to your empty canvas, then a *Transform Message Component*.

Author: ```suggestion
. Drag an HTTP Connector from the palette to your empty canvas, then a *Transform Message Component*.
```
